### ISSUE TITLE
merlin - The malicious bot drains the AutomationVault balance by executing the same job multiple times

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2024-04-xkeeper-judging#88]
merlin

high

# The malicious bot drains the AutomationVault balance by executing the same job multiple times

## Summary
The malicious bot drains the AutomationVault balance by executing the same job multiple times

## Vulnerability Detail
In the case of `GelatoRelay`, their bots are trusted due to how Gelato works, as stated in their README file. However, integrations with other protocols are restricted, so bots from `Keep3Relay` and `OpenRelay` are also restricted.

Let's consider a scenario with `OpenRelay` (the same problem would exist with `Keep3Relay`):

1)The user adds `OpenRelay`, approves callers (_ALL), and adds one JobData.
```solidity
 function addRelay(
    address _relay,
    address[] calldata _callers,
    IAutomationVault.JobData[] calldata _jobsData
  ) external onlyOwner {
```
2)A malicious approved caller calls `OpenRelay.exec` function with the same job multiple times. 
The `execData.length` cannot be 0, but it is possible to pass a length of up to 100 with the same job & job selector.
```solidity
function exec(
    IAutomationVault _automationVault,
    IAutomationVault.ExecData[] calldata _execData,
    address _feeRecipient
  ) external {
    if (_execData.length == 0) revert OpenRelay_NoExecData();  //
```
3)The malicious bot spends its gas, but since `OpenRelay` pays 120% for job execution, the malicious bot is 20% ahead in profit from the gas spent on execution. The `exec` function in `AutomationVault` only check whether job selector is approved.
```solidity
// Iterate over the exec data to execute the jobs
    for (_i; _i < _dataLength;) {
      _dataToExecute = _execData[_i];

      // Check that the selector is approved to be called
      if (!_approvedJobSelectors[msg.sender][_dataToExecute.job].contains(bytes4(_dataToExecute.jobData))) {
        revert AutomationVault_NotApprovedJobSelector();
      }
      (_success,) = _dataToExecute.job.call(_dataToExecute.jobData);
      if (!_success) revert AutomationVault_ExecFailed();

      // Emit the event
      emit JobExecuted(msg.sender, _relayCaller, _dataToExecute.job, _dataToExecute.jobData);

      unchecked {
        ++_i;
      }
    }
```

## Impact
The malicious bot drains the `AutomationVault` balance

## Code Snippet
[core/AutomationVault.sol#L409-L426](https://github.com/sherlock-audit/2024-04-xkeeper/blob/main/xkeeper-core/solidity/contracts/core/AutomationVault.sol#L409-L426)

## Tool used

Manual Review

## Recommendation

Consider allowing the execution of the same job with a job selector only once per transaction, otherwise revert. However, if a user has multiple jobs, a malicious bot can execute all jobs every time. Thus, consider allowing only one job per transaction for all approved callers and more than one job for a specific caller.

Duplicate of #4
